### PR TITLE
feat(tables-requirements): Add headers attribute requirement

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes
Add headers attribute requirement.

Screenshot:
image

#### Pull request checklist

- [x] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/content/test/semantics/headers-attribute.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        <h1>Headers attribute</h1>
+        <p>
+            The <Markup.Term>headers</Markup.Term> attribute of a <Markup.Term>{`<td>`}</Markup.Term> element must reference the correct{' '}
+            <Markup.Term>{`<th>`}</Markup.Term> element.
+        </p>
+
+        <h2>Why it matters</h2>
+        <p>
+            When assistive technologies describe data tables, they mention both the data cells and their programmatically-related header
+            cells. When incorrect header cells are reported, users are likely to find the table confusing. Proper use of the headers
```
Comment (RobGallo) [inline src/content/test/semantics/headers-attribute.tsx:16]: I think we should go back to Lisa Louise about this text. Either the word "proper" is wrong or the word "tedious" is wrong. Proper use would not make tables tedious. Either improper use would make tables tedious, or proper use would make tables easy to understand.

File: src/content/test/semantics/headers-attribute.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                Better: Eliminate multi-level headings and define headers using <Markup.Term>scope</Markup.Term> attributes.
+            </li>
+        </ul>
+        <p />
+
+        <h2>Example</h2>
+        <Markup.PassFail
+            failText={
+                <p>
+                    This complex data table has two levels of headers. Three of data cells' <Markup.Term>headers</Markup.Term> attributes
```
Comment (RobGallo) [inline src/content/test/semantics/headers-attribute.tsx:36]:
```suggestion
                    This complex data table has two levels of headers. Three of the data cells' <Markup.Term>headers</Markup.Term> attributes
```

File: src/content/test/semantics/headers-attribute.tsx (hunk 1 above)
Comment (haonliu) [reply src/content/test/semantics/headers-attribute.tsx:16]: Confirmed with Fer that we can remove the last sentence to unblock us.

File: src/content/test/semantics/headers-attribute.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        </ul>
+        <p />
+
+        <h2>Example</h2>
+        <Markup.PassFail
+            failText={
+                <p>
+                    This complex data table has two levels of headers. Three of the data cells' <Markup.Term>headers</Markup.Term>{' '}
+                    attributes should refer to both levels, but incorrectly refer to only the top-level header. Assistive technologies will
+                    announce only "Staff" as the header for those data cells. Some users will find it difficult to understand the table.
```
Comment (dbjorge) [inline src/content/test/semantics/headers-attribute.tsx:37]: In the passing example, the "Staff" header text was replaced by "Faculty"; the failing example should be consistent in the text label and here in this text

File: src/content/test/semantics/headers-attribute.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            <td headers="s">275</td>
+            <td [headers="f"]>10</td>
+            <td [headers="f"]>16</td>
+            <td [headers="f"]>5</td>
+            </tr>
+            </table> `}
+            passText={
+                <p>
+                    All data cells refer to the correct headers. Assistive technologies will announce both header levels where appropriate:
+                    "Staff Teachers," "Staff Assistants," and "Staff Administration". Everyone can interpret the table correctly.
```
Comment (dbjorge) [inline src/content/test/semantics/headers-attribute.tsx:60]: Same here (faculty, not staff)

File: src/content/test/semantics/headers-attribute.tsx (hunk 3 above)
Comment (haonliu) [reply src/content/test/semantics/headers-attribute.tsx:37]: Good catch! Checking with Fer the right word to use here.

Comment (haonliu) [reply src/content/test/semantics/headers-attribute.tsx:37]: Confirmed that they should both be Staff.

File: src/assessments/semantics/test-steps/headers-attribute.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                    <li>
+                        Each header cell (<Markup.CodeTerm>{`<th>`}</Markup.CodeTerm> element(s))
+                        must have an <Markup.Term>id</Markup.Term> attribute.
+                    </li>
+                    <li>
+                        Each data cell's <Markup.Term>headers</Markup.Term> attribute must reference
+                        all cells that function as headers for that data cell.
+                    </li>
+                </ol>
+                Note: If a <Markup.Term>headers</Markup.Term> attribute references an element(s)
```
Comment (dbjorge) [inline src/assessments/semantics/test-steps/headers-attribute.tsx:51]: This one can remain singular I think, the `(s)` is awkward with the "an"

File: src/content/test/semantics/headers-attribute.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { create, React } from '../../common';
+
+export const infoAndExamples = create(({ Markup, Link }) => (
+    <>
+        <h1>Headers attribute</h1>
+        <p>
+            The <Markup.Term>headers</Markup.Term> attribute of a <Markup.Term>{`<td>`}</Markup.Term> element must reference the correct{' '}
+            <Markup.Term>{`<th>`}</Markup.Term> element.
```
[the remark this answers is not in the document]
Comment (dbjorge) [reply src/content/test/semantics/headers-attribute.tsx:10]: Doesn't look like this one was actually fixed

File: src/content/test/semantics/headers-attribute.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        <p>
+            When assistive technologies describe data tables, they mention both the data cells and their programmatically-related header
+            cells. When incorrect header cells are reported, users are likely to find the table confusing.
+        </p>
+
+        <h2>How to fix</h2>
+        <ul>
+            <li>
+                Good: Modify the <Markup.Term>headers</Markup.Term> attribute of the <Markup.Term>{`<td>`}</Markup.Term> element to match
+                the <Markup.Term>id</Markup.Term> attribute of the correct <Markup.Term>{`<th>`}</Markup.Term> element, or
```
Comment (dbjorge) [inline src/content/test/semantics/headers-attribute.tsx:23]: element(s)

File: src/content/test/semantics/headers-attribute.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+            passText={
+                <p>
+                    All data cells refer to the correct headers. Assistive technologies will announce both header levels where appropriate:
+                    "Staff Teachers," "Staff Assistants," and "Staff Administration". Everyone can interpret the table correctly.
+                </p>
+            }
+            passExample={`<table border="1">
+            <tr>
+            <th rowspan="2" id="s">Students</th>
+            <th colspan="3" id="f">Staff</th>
```
Comment (dbjorge) [inline src/content/test/semantics/headers-attribute.tsx:66]: I think the "s" and "f" IDs made more sense when it was "Students" and "Faculty"; I think it's a little confusing to follow  the example now that the id ("f") has no relationship to the label, but I understand that the single-character IDs to fit horizontally in the example boxes. If we think Faculty is factually incorrect for "assistants", maybe keep "s"/"Staff" but replace "s"/Student" with "c"/"Class" for the example? (@ferBonnin)

File: src/content/test/semantics/headers-attribute.tsx (hunk 4 above)
Comment (dbjorge) [inline src/content/test/semantics/headers-attribute.tsx:60]: This says "Staff Administration" where the actual column label is "Staff Observers"

File: src/content/test/semantics/headers-attribute.tsx (hunk 8 above)
Comment (haonliu) [reply src/content/test/semantics/headers-attribute.tsx]: talked to Fer and decided to change all staff to faculty.

File: src/content/test/semantics/headers-attribute.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (haonliu) [reply src/content/test/semantics/headers-attribute.tsx]: Fixed

**Pull request merged**